code/_onclick/item_attack.dm

```diff
@@ -74,6 +74,7 @@ avoid code duplication. This includes items that may sometimes act as a standard
 // Proximity_flag is TRUE if this afterattack was called on something adjacent, in your square, or on your person.
 // Click parameters is the params string from byond Click() code, see that documentation.
 /obj/item/proc/afterattack(atom/target, mob/user, proximity_flag, params)
+	if(!proximity_flag) return // afterattack needs to check proximity_flag, because afterattack runs if you click at range. This is to fix invisible wrenches/hammers, and a general check. Unlikely to break anything.
```

Comment:
Change the afterattack of the objects in question instead of the parent proc for all objects.

Reply:
I can't, it's not handled that simply.

Reply:
It's handled in Click() where f it's not a gun it sends a false proximity flag. So this will always be false when it needs to be, it returns also, if it needs overriding somewhere it can be.

I was told to do this by a coderbus guy who said no check was incorrect but everything else in the click chain is working as intended.